encrypted-media-respec.html

```diff
@@ -750,73 +798,63 @@
                 </dl>
               </li>
 
-              <li><p>If <var>implementation</var> in the configuration specified by the combination of the values in <var>accumulated configuration</var> is not supported or not allowed in the <var>origin</var>, return <code>null</code>.</p>
+              <li><p>If <var>implementation</var> in the configuration specified by the combination of the values in <var>accumulated configuration</var> is not supported or not allowed in the <var>origin</var>, return <code>NotSupported</code>.</p>
                 <p class="note">In this step, "supported" includes the implementation being available for use when this algorithm returns, not just user agent support for such an implementation.</p>
               </li>
 
-              <li><p>If <var>accumulated configuration</var>'s <a def-id="option-distinctiveIdentifier"></a> value is <a def-id="requirement-required"></a>:</p>
+              <li>
+                <p>
+                  If <var>accumulated configuration</var>'s <a def-id="option-distinctiveIdentifier"></a> value is <a def-id="requirement-required"></a> and if
+                  the <a def-id="distinctive-identifier"></a> is not <a href="#per-origin-identifiers">unique per-origin</a> and
+                  <a href="#allow-identifiers-cleared">clearable</a>:
+                </p>
                 <ol>
                   <li>
+                    <p>Add the configurations described by <var>accumulated configuration</var> to <var>restrictions</var>.</p>
```

Comment:
This one is odd, because it is a requirement that is relevant only for non-compliant implementations and so one wonders why non-compliant implementations would take the trouble to be compliant to this.

Anyway, _restrictions_ records all the values of _accumulated configuration_ for which the user has denied consent (or, at least, it is a rule which can determine whether a given specific value of _accumulated configuration_ has been denied consent - see above). In this case, the specification is effectively denying consent on behalf of the user, so  it makes sense to update _restrictions_. If we don't won't we just come straight back around to here again ?

When we add _accumulated configuration_ to restrictions we add that single combination of parameters, codecs, robustness etc. We may well come around again with one that is only very slightly different, but eventually we'll run out of options that trigger this check and take a different path.